### PR TITLE
SL-17597 AV height in Shape dialog IS WRONG

### PR DESCRIPTION
**Pull request by LLGuru**

### CODE REVIEW

**Review by marchcat-pe — APPROVED** [review]
This looks good,
but should be tested on smaller avatars (dinkies, tinies, etc.)

**Review by akleshchev — APPROVED** [review]

**Comment by akleshchev** [comment]
Looks much cleaner than the original.

**Comment by akleshchev** [comment]
https://github.com/secondlife/jira-archive-internal/issues/67771

**Comment by marchcat** [comment]
> should be tested on smaller avatars (dinkies, tinies, etc.)

+ ponies.

File: indra/llappearance/llavatarappearance.cpp [hunk 1]
```diff
@@ -508,70 +508,45 @@ void LLAvatarAppearance::computeBodySize()
...
+}
+
+F32 LLAvatarAppearance::computePelvisToFoot()
+{
+    F32 result =
+        // all these relative positions usually are negative
+        mPelvisp->getScale().mV[VZ] * mHipLeftp->getPosition().mV[VZ] +
+        mHipLeftp->getScale().mV[VZ] * mKneeLeftp->getPosition().mV[VZ] +
+        mKneeLeftp->getScale().mV[VZ] * mAnkleLeftp->getPosition().mV[VZ] +
+        mAnkleLeftp->getScale().mV[VZ] * mFootLeftp->getPosition().mV[VZ] / 2;
```
Comment (akleshchev) [inline indra/llappearance/llavatarappearance.cpp:546]: This leaves soma avatars' foots below surface, can you explain why you changed it like this?

**Pull request merged**